components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java

```diff
@@ -231,6 +244,74 @@ public boolean doPreSetUserClaimValuesWithID(String userID, Map<String, String>
         return true;
     }
 
+    public boolean doPreSetUserClaimValues(String userName, Map<String, String> claims, String profileName,
+                                           UserStoreManager userStoreManager) throws UserStoreException {
+
+        // Validate whether claim update request is for a provisioned user.
+        if (isJitProvisionEnhancedFeature()) {
+            validateJitProvisionedUserClaimUpdate(userName, profileName, userStoreManager);
+        }
+        return true;
+    }
+
+    public boolean doPreSetUserClaimValue(String userName, String claimURI, String claimValue, String profileName,
+                                          UserStoreManager userStoreManager) throws UserStoreException {
+
+        // Validate whether claim update request is for a provisioned user.
+        if (isJitProvisionEnhancedFeature()) {
+            validateJitProvisionedUserClaimUpdate(userName, profileName, userStoreManager);
+        }
+        return true;
+    }
+
+    private String getUsernameFromUserID(String userID, UserStoreManager userStoreManager) throws UserStoreException {
+
+        return ((AbstractUserStoreManager) userStoreManager).getUserNameFromUserID(userID);
+    }
+
+    /**
+     * Validate whether the claim update request is from a provisioned user.
+     *
+     * @param username         Username.
+     * @param profileName      Profile name.
+     * @param userStoreManager Userstore manager.
+     * @throws UserStoreException if an error occurred while retrieving the user claim list.
+     */
+    private void validateJitProvisionedUserClaimUpdate(String username, String profileName,
+                                                       UserStoreManager userStoreManager) throws UserStoreException {
+
+        /*
+        Check whether the config is enabled to edit synced attributes of the provisioned user.
+        If it is enabled, blocking the attribute editing is not required.
+         */
+        if (!Boolean.parseBoolean(IdentityUtil.getProperty(ENABLE_JIT_PROVISION_ENHANCE_FEATURE))) {
+            return;
+        }
+
+        /*
+        Check whether this is an attribute syncing flow by checking the PROVISIONED_USER thread local property.
+        If it is an attribute syncing flow, blocking the attribute editing is not required.
+         */
+        if (IdentityUtil.threadLocalProperties.get().get(FrameworkConstants.JIT_PROVISIONING_FLOW) != null &&
+                (Boolean) IdentityUtil.threadLocalProperties.get().get(FrameworkConstants.JIT_PROVISIONING_FLOW)) {
+            return;
+        }
+
+        boolean isExistingJITProvisionedUser;
```

Comment:
Although we don't show syncEnabled properly in UI level, can we have a property in BE and check this here? ideally this should happen only if we enabled sync from IDP.

Reply:
Fixed https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/367/commits/615427cb98805016d37d6068432a2f558703d653